### ISSUE TITLE
Should the applicability extent be defined by a WKT multipolygon

### ISSUE THREAD

**Issue by ccrook** [opengeospatial/CRS-Gridded-Geodetic-data-eXchange-Format#11]
GGXF has two bounding boxes:

contentBox defines total extent of grid
contentApplicabilityBox defines area of applicability, ie where it is reasonable to use, or where it is authoritative.

Suggestion is that second should include option of WKT multipolygon definition to correctly define area in which the producer intends that the model should be used.  May optionally be used by software to advise user of reliability/applicability of usage.

**Comment by ccrook** [comment]
KR: Should this be in the same file, should it be in GeoPackage.
Various: Concern about size of grid

**Comment by ccrook** [comment]
RL: Conclusion - will add polyhgon options

**Comment by ccrook** [comment]
Re-opened for discussion on follow up meeting (in two days time).  Also to add a couple of notes:

1. The polygon format need not be WKT, indeed within the GGXF binary format it would likely be a binary format (in the meeting chat someone - not sure who - did suggest it could be WKB).  
2. Concern about the potential size of a polygon definition within a header is certainly valid.  It could make it much slower to read the file header, which might make it manageable.  Depending on the transport format for GGXF it may be possibly to separate the definition of the polygon from the actual header and some form of internal pointer to identify its location in the binary representation.  In most practical cases I believe the size of the polygon will be a relatively small compared to the size of the grids in the GGXF.   Also it is at the producers discretion to decide how accurately they wish represent the extent of applicability, being cognisant of the impact on file size.  If the producer sees some value in a very precise polygon extent for a small file, that is their choice.  If the are concerned that a polygon is making the file too big, they can simplify it.
3. I believe the primary purpose of applicability extent is discovery.  It is used to assess whether or not a user might choose the GGXF for their purpose.  
4. We might specify how calculation software is expected to use it.  Options could include notify the user, fail the calculation if it outside the extent, and totally ignore it.  My inclination is that use of the polygon, if present, would be optional.  The system may choose to warn the user (even by failing the calculation), but this would not be a requirement of compliant software.  
5. The header should include both a bounding box as well as a optional polygon for defining the extent of applicability to allow systems that cannot use polygons to ignore it.
6. I question the value of a contentBox (or contentExtent) as well as a applicabilityExtent, as an ISO compliant lat/lon extent.  The only value I see in the contentExtent is to define whether or not the GGXF _can_ be calculated in the extent.  The applicability extent determines whether it should be used.  Necessarily the applicabilityExtent is a subset of the contentExtent. 
 From a discovery point of view there is no reason to advertise that your deformation model (or whatever) can be used where the values it generates are simply there to fill the model out to a rectangular grid.  
7. To define whether a GGXF _can_ be used at an extent requires extent to be in the interpolation CRS.  Each group has its own extent (which in 21 June meeting we decided would be implicitly defined by affine parameters and row/column counts).  For  most GGXF, with a single group, the contentExtent would be the same as the group extent.   For a multiple group GGXF it may add value as a union of the group extents to avoid potentially wasteful calculations testing each group extent for a point outside all the groups.   However there is no requirement that the groups together fill rectangular area - they could be discrete or overlapping areas - so again the contentExtent is potentially not adding much value beyond the applicability extent.

In summary my opinion is that  the optimal specification includes a contentApplicability bounding box and optional polygon extent in the file header, and does not include file or group content extent(s).   When we look at implementation in a specific carrier we may choose to add the to the header for computational efficiency, but they are not part of the logical specification of the standard or the text format implementing it.

**Comment by ccrook** [comment]
The proposal for an optional polygon bounds for contentApplicability is implemented in the GGXF specification. Closing this issue.
